server/services/connection-service.js

```diff
@@ -54,12 +56,21 @@ module.exports = {
     } catch (err) {
       console.log('unable to close connection', err)
     }
+  },
+  isConnected: async function () {
+    if (connections.length !== 2) {
+      return false
+    }
+    connections.forEach(connection => {
+      if (!connection.isOpen()) {
+        return false
+      }
+    })

```

Comment:
This foreach might not do what you think it does - the return is only from the inner foreach loop, and will not return false from the outer isConnected function.

It'll only return false if there are not 2 connections. This isn't caught in the test as the test is stubbing the isConnected function's return value and not calling the internal logic.

Reply:
Refactored the logic.